SDD-785 rename SG RF2 export files

AAAlinaaa opened the pull request.

AAAlinaaa committed:
SDD-785 Rename SG RF1 and RF2 export files

AAAlinaaa committed:
SDD-785 Rename export files in crossmap and subsetmember exporter too

AAAlinaaa committed:
SDD-785 Avoid circular dependency in snomed exporter

nagyo reviewed (changes requested):
I'm seeing the following issues here:
This change should result in proper filenames for:
- the SG client (SG1000132 or SG1000133)
- the nebraskan client (US1000004 or US1000124)
- the GA client if needed (INT or any configured country, namespace pair)
The language code should be extracted by using com.b2international.snowowl.snomed.core.lang.LanguageSetting, the default namespace should be extracted by using the default namespace configuration of the client (com.b2international.snowowl.snomed.datastore.SnomedConfiguration)

Some hints for the fixes:
- DRY (don't repeat yourself) principle could be satisfied by moving the getCountryAndNameSpaceElement() method to e.g. com.b2international.snowowl.snomed.datastore.internal.rf2.SnomedExporterUtil
- the default client namespace is already propagated through the export process com.b2international.snowowl.snomed.exporter.server.SnomedExporterFacade.clientNamespace

AAAlinaaa committed:
SDD-785 Adapt for use for countries other than SG, make ...

... countryAndNamespaceElement available from export configuration

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter/src/com/b2international/snowowl/snomed/exporter/model/SnomedExporterUtil.java:
```diff
@@ -95,4 +98,18 @@ private static boolean isConceptType(final SnomedRefSetIndexEntry entry) {
+	public static String getCountryAndNameSpaceElement() {
+		String namespace = getNamespace();
+		String countryCode = getCountry().toUpperCase();
+		return  namespace.isEmpty() ? "_INT_" : "_" + countryCode + namespace + "_";
```
Use Strings.isNullOrEmpty() for checking the namespace emptiness.

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/net4j/SnomedExportServerIndication.java:
```diff
@@ -94,7 +94,7 @@
 	private Set<String> modulesToExport;
 	private Date deltaExportStartEffectiveTime;
 	private Date deltaExportEndEffectiveTime;
-	private String clientNamespace;
```
I see the following filenames in the archive when I initiate an export through the REST api:

![image](https://user-images.githubusercontent.com/2010200/30649204-fa972060-9e1f-11e7-9d9c-e8890b44e090.png)

nagyo replied:
The point is that this field is called clientNamespace not just by coincidence.

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter/src/com/b2international/snowowl/snomed/exporter/model/SnomedRf2ExportModel.java:
```diff
@@ -291,25 +287,14 @@ public void setRefSetsToExport(final boolean refSetsToExport) {
 		final String token;
 		if (!singleRefSetExport) {
-			token = new StringBuilder().append("SnomedCT_Release").append(getCountryAndNameSpaceElement()).append(Dates.formatByHostTimeZone(new Date(), "yyyyMMdd-HHmm")).toString();
+			token = new StringBuilder().append("SnomedCT_Release").append(namespace).append(Dates.formatByHostTimeZone(new Date(), "yyyyMMdd-HHmm")).toString();
```
I see the following root folder inside the export archive (result of delta export using the client):
![image](https://user-images.githubusercontent.com/2010200/30649355-5b801c74-9e20-11e7-89ec-ab43f0980c4b.png)

AAAlinaaa committed:
SDD-785: Include changes requested in code review

1. Rename namespace variable in REST export
2. Ensure folder namings doesn't have extra "_"s
3. Include null check for namespace in Export Utils

nagyo commented on snomed/com.b2international.snowowl.snomed.api.impl/src/com/b2international/snowowl/snomed/api/impl/domain/SnomedExportConfiguration.java:
```diff
@@ -43,7 +43,7 @@ public SnomedExportConfiguration(Rf2ReleaseType type,
 			Date deltaExportStartEffectiveTime, Date deltaExportEndEffectiveTime, 
```
Please rename the method parameter to reflect the content as well!

nagyo commented on snomed/com.b2international.snowowl.snomed.api.impl/src/com/b2international/snowowl/snomed/api/impl/SnomedExportService.java:
```diff
@@ -96,7 +96,7 @@ private SnomedRf2ExportModel convertConfiguration(final ISnomedExportConfigurati
 		final SnomedRf2ExportModel model = createExportModelWithAllRefSets(contentSubType, exportBranch);
 		
-		final String namespaceId = configuration.getNamespaceId();
+		final String namespaceId = configuration.getCountryAndNamespaceId();
```
Confusing variable name!

nagyo commented on snomed/com.b2international.snowowl.snomed.api.impl/src/com/b2international/snowowl/snomed/api/impl/SnomedExportService.java:
```diff
@@ -96,7 +96,7 @@ private SnomedRf2ExportModel convertConfiguration(final ISnomedExportConfigurati
 		
-		final String namespaceId = configuration.getNamespaceId();
+		final String namespaceId = configuration.getCountryAndNamespaceId();
 		model.setNamespace(namespaceId);
```
This should also be clear what we are setting here.

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter/src/com/b2international/snowowl/snomed/exporter/model/SnomedRf2ExportModel.java:
```diff
@@ -168,6 +168,7 @@ public SnomedRf2ExportModel(@Nullable final String refSetId) {
 		modulesToExport = Sets.newHashSet();
 		userId = ApplicationContext.getInstance().getService(ICDOConnectionManager.class).getUserId();
 		unsetEffectiveTimeLabel = "";
+		namespace = SnomedExporterUtil.getCountryAndNameSpaceElement();
```
Inconsistent variable name.

nagyo reviewed (changes requested):
I see "clientNamespace" or "namespace fields in:
- com.b2international.snowowl.snomed.exporter.server.SnomedExporterFacade
- com.b2international.snowowl.snomed.exporter.server.SnomedExportExecutor
- com.b2international.snowowl.snomed.exporter.service.SnomedExportService
- com.b2international.snowowl.snomed.exporter.model.SnomedRf2ExportModel

These are all ambiguous.

I see the following description file name in a delta export result archive: sct2_Description_Delta-*en-sg*_SG1000132_20170925.txt The country-namespace tag looks OK, but the language code does not, please fix that as well.

AAAlinaaa committed:
SDD-785: Ensure consistent variable names for countryAndNamespaceId

AAAlinaaa committed:
SDD-785: Ensure consistent variable names for countryAndNamespaceId v2

AAAlinaaa committed:
SDD-785: Convert country code to upper case in langua code elements...

... of export filenames
